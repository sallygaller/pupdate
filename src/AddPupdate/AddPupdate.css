.AddPupdate h2 {
    text-align: center;
}

.AddPupdate {
    border-radius: 5px;
    background-color: #f2f2f2; 
}

.AddPupdate-form {
    display: flex;
    flex-direction: column;
    margin: 0 auto;
    width: 50%;
}

.AddPupdate textarea {
    font-family: sans-serif
}

.AddPupdate-error {
    color: red;
    text-align: center;
} 

.AddPupdate input {
    width: 100%;
    padding: 12px 20px;
    margin: 8px 0;
    display: inline-block;
    border: 1px solid #ccc;
    border-radius: 4px;
    box-sizing: border-box;
} 

.AddPupdate label {
    margin-bottom: 5px;
}

.AddPupdate button {
    width: 100%;
    border: #F2AD1D solid 2px;
    border-radius: 25px;
    padding: 14px 20px;
    margin: 8px 0;
    cursor: pointer;
}

.AddPupdate button:hover {
    background-color: #F2AD1D;
}
