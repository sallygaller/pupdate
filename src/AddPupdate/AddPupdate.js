import React, { useState } from "react";
import { useHistory } from "react-router";
import SearchLocationInput from "../SearchLocation/SearchLocation";
import TokenService from "../services/token-service";
import { API_ENDPOINT } from "../config";
import "./AddPupdate.css";

export default function AddPupdate() {
  const history = useHistory();
  const [date, setDate] = useState("");
  const [starttime, setStarttime] = useState("");
  const [endtime, setEndtime] = useState("");
  const [location, setLocation] = useState("");
  const [description, setDescription] = useState("");
  const [error, setError] = useState(null);

  const Required = () => <span>*</span>;

  const handleSubmit = (e) => {
    e.preventDefault();
    const pupdate = {
      date,
      starttime,
      endtime,
      location,
      description,
    };
    fetch(API_ENDPOINT + `/pupdates`, {
      method: "POST",
      body: JSON.stringify(pupdate),
      headers: {
        "content-type": "application/json",
        authorization: `bearer ${TokenService.getAuthToken()}`,
      },
    })
      .then((res) => {
        if (!res.ok) {
          return res.json().then((error) => {
            throw error;
          });
        }
        return res.json();
      })
      .then((data) => {
        history.push("/pupdates");
      })
      .catch((error) => {
        setError(error.message);
      });
  };

  const handleClickCancel = () => {
    history.push("/pupdates");
  };

  return (
    <div className="AddPupdate">
      <h2>New Pupdate</h2>
      <form className="AddPupdate-form" onSubmit={(e) => handleSubmit(e)}>
        <div className="AddPupdate-error" role="alert">
          {error && (
            <p>
              {error.message}
              {error}
            </p>
          )}
        </div>
        <label htmlFor="location">
          Location:
          <Required />
        </label>
        <SearchLocationInput
          required
          location={location}
          setLocation={setLocation}
          placeholder={"Enter a location (e.g. a local park)"}
        />
        <label htmlFor="date">
          Date:
          <Required />
        </label>
        <input
          required
          id="date"
          type="date"
          value={date}
          onChange={(e) => setDate(e.target.value)}
        />
        <label htmlFor="starttime">
          Start Time:
          <Required />
        </label>
        <input
          required
          id="starttime"
          type="time"
          value={starttime}
          onChange={(e) => setStarttime(e.target.value)}
        />
        <label htmlFor="endtime">
          End Time
          <Required />{" "}
        </label>
        <input
          required
          id="endtime"
          type="time"
          value={endtime}
          onChange={(e) => setEndtime(e.target.value)}
        />
        <label htmlFor="description">Description:</label>
        <textarea
          id="description"
          name="description"
          className="AddPupdate-textarea"
          rows="5"
          cols="10"
          value={description}
          onChange={(e) => setDescription(e.target.value)}
        />
        <button type="submit">Submit</button>{" "}
        <button type="button" onClick={handleClickCancel}>
          Cancel
        </button>
      </form>
    </div>
  );
}
