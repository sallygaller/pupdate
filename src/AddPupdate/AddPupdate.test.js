import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import AddPupdate from "./AddPupdate";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <AddPupdate />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer.create(<AddPupdate name="AddPupdate" />).toJSON();
  expect(tree).toMatchSnapshot();
});
