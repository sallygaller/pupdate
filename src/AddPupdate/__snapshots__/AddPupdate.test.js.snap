// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="AddPupdate"
>
  <h2>
    New Pupdate
  </h2>
  <form
    className="AddPupdate-form"
    onSubmit={[Function]}
  >
    <div
      className="AddPupdate-error"
      role="alert"
    />
    <label
      htmlFor="location"
    >
      Location:
      <span>
        *
      </span>
    </label>
    <div
      className="search-location-input"
    >
      <input
        onChange={[Function]}
        placeholder="Enter a location (e.g. a local park)"
        value=""
      />
    </div>
    <label
      htmlFor="date"
    >
      Date:
      <span>
        *
      </span>
    </label>
    <input
      id="date"
      onChange={[Function]}
      required={true}
      type="date"
      value=""
    />
    <label
      htmlFor="starttime"
    >
      Start Time:
      <span>
        *
      </span>
    </label>
    <input
      id="starttime"
      onChange={[Function]}
      required={true}
      type="time"
      value=""
    />
    <label
      htmlFor="endtime"
    >
      End Time
      <span>
        *
      </span>
       
    </label>
    <input
      id="endtime"
      onChange={[Function]}
      required={true}
      type="time"
      value=""
    />
    <label
      htmlFor="description"
    >
      Description:
    </label>
    <textarea
      className="AddPupdate-textarea"
      cols="10"
      id="description"
      name="description"
      onChange={[Function]}
      rows="5"
      value=""
    />
    <button
      type="submit"
    >
      Submit
    </button>
     
    <button
      onClick={[Function]}
      type="button"
    >
      Cancel
    </button>
  </form>
</div>
`;
